rust/cubesql/cubesql/src/compile/mod.rs

```diff
@@ -4048,94 +4072,169 @@ ORDER BY \"COUNT(count)\" DESC"
             (
                 "COUNT(*), DATE(order_date) AS __timestamp".to_string(),
                 "order_date >= STR_TO_DATE('2021-08-31 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f') AND order_date < STR_TO_DATE('2021-09-07 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')".to_string(),
-                Some(vec![V1LoadRequestQueryTimeDimension {
-                    dimension: "KibanaSampleDataEcommerce.order_date".to_string(),
-                    granularity: Some("day".to_string()),
-                    date_range: Some(json!(vec![
-                        "2021-08-31T00:00:00.000Z".to_string(),
-                        "2021-09-06T23:59:59.999Z".to_string()
-                    ])),
-                }])
+                Some(vec![
+                    V1LoadRequestQueryFilterItem {
+                        member: Some("KibanaSampleDataEcommerce.order_date".to_string()),
+                        operator: Some("afterOrOnDate".to_string()),
+                        values: Some(vec!["2021-08-31T00:00:00.000Z".to_string()]),
+                        or: None,
+                        and: None,
+                    },
+                    V1LoadRequestQueryFilterItem {
+                        member: Some("KibanaSampleDataEcommerce.order_date".to_string()),
+                        operator: Some("beforeDate".to_string()),
+                        values: Some(vec!["2021-09-06T23:59:59.999Z".to_string()]),
```

Comment:
There's some code that is forcing this interval to change and I cannot seem to trace down to see where this is occuring.

Reply:
It's a regression. Please see further comments.